tests/log_test.h

```diff
@@ -0,0 +1,177 @@
+/*
+* Copyright 2010-2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+*
+* Licensed under the Apache License, Version 2.0 (the "License").
+* You may not use this file except in compliance with the License.
+* A copy of the License is located at
+*
+*  http://aws.amazon.com/apache2.0
+*
+* or in the "license" file accompanying this file. This file is distributed
+* on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+* express or implied. See the License for the specific language governing
+* permissions and limitations under the License.
+*/
+
+#include <aws/testing/aws_test_harness.h>
+#include <aws/common/atomic.h>
+#include <aws/common/log.h>
+
+static void log_report_fn(const char *log_message) {
+    fprintf(AWS_TESTING_REPORT_FD, "%s", log_message);
+}
+
+AWS_TEST_CASE(test_log_init_clean_up, test_log_init_clean_up_fn)
+static int test_log_init_clean_up_fn(struct aws_allocator *allocator, void *ctx) {
+    ASSERT_SUCCESS(aws_log_system_init(allocator, 1024, 256, 10));
+    aws_log_set_reporting_callback(log_report_fn);
+
+    ASSERT_SUCCESS(aws_log(AWS_LOG_LEVEL_TRACE, "Oh, hello there #1."));
+    ASSERT_SUCCESS(aws_log_flush());
+    ASSERT_SUCCESS(aws_log(AWS_LOG_LEVEL_TRACE, "Oh, hello there #2."));
+    ASSERT_SUCCESS(aws_log_flush());
+    ASSERT_SUCCESS(aws_log(AWS_LOG_LEVEL_TRACE, "Oh, hello there #3."));
+    ASSERT_SUCCESS(aws_log(AWS_LOG_LEVEL_TRACE, "Oh, hello there #4."));
+    ASSERT_SUCCESS(aws_log(AWS_LOG_LEVEL_TRACE, "Oh, hello there #5."));
+    ASSERT_SUCCESS(aws_log_flush());
+
```

Comment:
We don't assert that the logs actually made it through? [in this, as well as test_log_overflow_message_fn, and test_log_threads_hammer_fn]

Reply:
Not asserted, but were placed on the console.

Reply:
Tests should assert whatever they're testing; don't rely on manual observation of test output (this output is usually suppressed by CTest anyway).